components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementFacade.java

```diff
@@ -0,0 +1,173 @@
+            throws AuthenticatorMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            return jdbcTemplate.withTransaction(template -> {
+                List<UserDefinedLocalAuthenticatorConfig> configList =
+                        dao.getAllUserDefinedLocalAuthenticators(tenantId);
+                for (UserDefinedLocalAuthenticatorConfig config : configList) {
+                    endpointConfigManager.resolveEndpointConfigurations(config, tenantId);
+                }
+                return configList;
+            });
+        } catch (TransactionException e) {
+            // Since exceptions thrown are wrapped with TransactionException, extracting the actual cause.
+            LOG.debug("Error while retrieving all user defined local authenticators in Tenant Domain: " +
+                    IdentityTenantUtil.getTenantDomain(tenantId));
+            throw handleAuthenticatorMgtException(e.getCause());
+        }
+    }
+
+    @Override
+    public void deleteUserDefinedLocalAuthenticator(String authenticatorConfigName, UserDefinedLocalAuthenticatorConfig
+            authenticatorConfig, int tenantId) throws AuthenticatorMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                endpointConfigManager.deleteEndpointConfigurations(authenticatorConfig, tenantId);
+                dao.deleteUserDefinedLocalAuthenticator(authenticatorConfigName, authenticatorConfig, tenantId);
+                return null;
+            });
+        } catch (TransactionException e) {
+            // Since exceptions thrown are wrapped with TransactionException, extracting the actual cause.
+            LOG.debug("Error while deleting the user defined local authenticator: " + authenticatorConfigName +
+                    " in Tenant Domain: " + IdentityTenantUtil.getTenantDomain(tenantId) +
+                    ". Rolling back deleted authenticator information, and associated action.");
+            throw handleAuthenticatorMgtException(e.getCause());
+        }
+    }
+
+    /**
+     * Handle the authenticator management client exception.
+     *
+     * @param throwable Throwable object.
+     * @throws AuthenticatorMgtClientException If an authenticator management client exception.
+     */
+    private static AuthenticatorMgtException handleAuthenticatorMgtException(Throwable throwable)
+            throws AuthenticatorMgtException {
+
+        if (throwable instanceof AuthenticatorMgtClientException) {
+            AuthenticatorMgtClientException error = (AuthenticatorMgtClientException) throwable;
+            throw new AuthenticatorMgtClientException(error.getErrorCode(), error.getMessage(), error.getDescription());
+        }
```

Comment:
Better add a comment mentioning the reason to handle these client error failures